pkg/storage/snapshot/source.go

```diff
@@ -407,6 +407,55 @@ func (s *vmSnapshotSource) Spec() (snapshotv1.SourceSpec, error) {
 	}, nil
 }
 
+func (s *vmSnapshotSource) handleVolumes(vmCpy *snapshotv1.VirtualMachine) error {
+	vmi, exists, err := s.controller.getVMI(s.vm)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("can't get vm revision, vmi doesn't exist")
+	}
+
+	vmVolMap := map[string]kubevirtv1.Volume{}
+	for _, vol := range s.vm.Spec.Template.Spec.Volumes {
+		vmVolMap[vol.Name] = vol
+	}
+
+	vmDiskMap := map[string]kubevirtv1.Disk{}
+	for _, disk := range s.vm.Spec.Template.Spec.Domain.Devices.Disks {
+		if _, ok := vmVolMap[disk.Name]; ok {
+			vmDiskMap[disk.Name] = disk
+		}
+	}
+
+	vmiVolNames := sets.New[string]()
+	for _, vol := range vmi.Spec.Volumes {
+		vmiVolNames.Insert(vol.Name)
+	}
+
+	vmCpyVolNames := sets.New[string]()
+	for _, vol := range vmCpy.Spec.Template.Spec.Volumes {
```

Comment:
not sure why you even need to revision volumes? dont we just want all the volume that are currently in the VMI which are also persistent? so its just enough to check if the volumes are both in the current vm and in the vmi